src/IO/ZstdDeflatingWriteBuffer.cpp

```diff
@@ -84,6 +85,9 @@ void ZstdDeflatingWriteBuffer::nextImpl()
 
 void ZstdDeflatingWriteBuffer::finalizeBefore()
 {
+    /// Don't write out if no data was ever compressed
```

Comment:
Could you fix that behaviour in ut-test? It seems important trait.

Reply:
not quite sure what you are talking about - this change is only applicable to some specific cases within this PR changes...

Reply:
I read it as do not even write a file if it is empty, is that correct?

Reply:
yes, but it's not about file, it's about compressed http data - we don't want to have compressed http body with no data

Reply:
Ok.